internal/command/instance.go

```diff
@@ -346,6 +352,35 @@ func (c *Commands) SetUpInstance(ctx context.Context, setup *InstanceSetup) (str
 		)
 	}
 
+	accessTokenLifetime := time.Duration(0)
+	idTokenLifetime := time.Duration(0)
+	refreshTokenIdleExpiration := time.Duration(0)
+	refreshTokenExpiration := time.Duration(0)
+
+	if setup.OIDCSettings != nil {
+		if setup.OIDCSettings.AccessTokenLifetime != time.Duration(0) {
+			accessTokenLifetime = setup.OIDCSettings.AccessTokenLifetime
+		}
+		if setup.OIDCSettings.IdTokenLifetime != time.Duration(0) {
+			idTokenLifetime = setup.OIDCSettings.IdTokenLifetime
+		}
+		if setup.OIDCSettings.RefreshTokenIdleExpiration != time.Duration(0) {
+			refreshTokenIdleExpiration = setup.OIDCSettings.RefreshTokenIdleExpiration
+		}
+		if setup.OIDCSettings.RefreshTokenExpiration != time.Duration(0) {
+			refreshTokenExpiration = setup.OIDCSettings.RefreshTokenExpiration
+		}
+	}
+	validations = append(validations,
+		c.prepareAddOIDCSettings(
+			instanceAgg,
+			accessTokenLifetime,
+			idTokenLifetime,
+			refreshTokenIdleExpiration,
+			refreshTokenExpiration,
+		),
+	)
```

Comment:
shouldn't the append of the validation only occur if the settings are not nil? otherwise the pointer could be removed as the durations are checked for default values anyway

i'd also then change the declarations to var as it might not be necessary to initialise them at all (var accessTokenLifetime time.Duration)
and the validation will fail if any of the durations is 0.. so should be check them here already or at least have a note in the defaults.yml that either every duration has to be set or none

Reply:
Changed it with a comment in the default.yaml, and that the event will only be created if the pointer != nil